packages/story-editor/src/components/canvas/extraPages.js

```diff
@@ -0,0 +1,161 @@
+  align-items: center;
+  height: 100%;
+`;
+const ExtraPageList = styled.ol`
+  display: flex;
+  flex-direction: ${({ isPrevious }) => (isPrevious ? 'row-reverse' : 'row')};
+  width: ${({ listWidth }) => listWidth}px;
+  height: ${({ extraPageHeight }) => extraPageHeight}px;
+  margin: 0;
+  padding: 0 ${GAP}px;
+  gap: ${GAP}px;
+`;
+const ExtraPage = styled.li`
+  display: block;
+  width: ${({ extraPageWidth }) => extraPageWidth - GAP}px;
+  height: 100%;
+  border-radius: 4px;
+  background-color: white;
+
+  opacity: 0.5;
+  transition: opacity 0.2s ease;
+  &:hover {
+    opacity: 1;
+  }
+`;
+const ExtraPagePreview = styled(PagePreview)`
+  cursor: pointer;
+`;
+
+function range(from, to) {
+  return Array.from(Array(to - from)).map((k, v) => v + from);
+}
+
+function getPagesToShow({
+  isPrevious,
+  hasExtraPages,
+  currentPageIndex,
+  extraPageCount,
+  pageCount,
+}) {
+  if (!hasExtraPages) {
+    return [];
+  }
+  // Showing pages before the current one
+  if (isPrevious) {
+    const from = Math.max(0, currentPageIndex - extraPageCount);
+    const to = currentPageIndex;
+    return from < to ? range(from, to).reverse() : [];
+  }
+  // Showing pages after the current one
+  const from = Math.min(pageCount - 1, currentPageIndex + 1);
+  const to = Math.min(pageCount - 1, currentPageIndex + extraPageCount + 1);
+  return from < to ? range(from, to) : [];
+}
+
+function ExtraPages({ isPrevious = false }) {
+  const { currentPageIndex, pages, setCurrentPage } = useStory(
+    ({ state: { pages, currentPageIndex }, actions: { setCurrentPage } }) => ({
+      currentPageIndex,
+      pages,
+      setCurrentPage,
+    })
+  );
+  const { hasExtraPages, extraPageWidth, extraPageCount } = useLayout(
+    ({ state: { hasExtraPages, extraPageWidth, extraPageCount } }) => ({
+      hasExtraPages,
+      extraPageWidth,
+      extraPageCount,
+    })
+  );
+  const isExtraPagesEnabled = useFeature('extraPages');
+  const pageCount = pages?.length;
+  if (!pageCount || !isExtraPagesEnabled) {
+    return null;
+  }
+  const pagesToShow = getPagesToShow({
+    isPrevious,
+    hasExtraPages,
+    currentPageIndex,
+    extraPageCount,
+    pageCount,
+  });
+  if (pagesToShow.length === 0) {
+    return null;
+  }
+  const listWidth = pagesToShow.length * extraPageWidth;
+  const extraPageHeight = (extraPageWidth - GAP) / PAGE_RATIO;
+
+  const clickPage = (pageId) => () => setCurrentPage({ pageId });
+
+  return (
+    <ExtraPageWrapper isPrevious={isPrevious}>
+      <ExtraPageList
+        isPrevious={isPrevious}
+        listWidth={listWidth}
+        extraPageHeight={extraPageHeight}
+      >
+        {pagesToShow.map((pageNum) => (
+          <ExtraPage key={pageNum} extraPageWidth={extraPageWidth}>
+            <ExtraPagePreview
+              page={pages[pageNum]}
+              onClick={clickPage(pages[pageNum].id)}
+              aria-label={sprintf(
+                /* translators: %s: Page number */
```

Comment:
```suggestion
                /* translators: %s: page number. */
```

To address the i18n lint warning

> Warning: The string "Go to page %s" has 2 different translator comments.